src/ZODB/tests/testblob.py

```diff
@@ -60,14 +51,17 @@ def new_time():
     """Create a _new_ time stamp.
 
     This method also makes sure that after retrieving a timestamp that was
-    *before* a transaction was committed, that at least one second passes so
+    *before* a transaction was committed, that at least one half second passes so
     the packing time actually is before the commit time.
 
     """
     now = new_time = time.time()
     while new_time <= now:
         new_time = time.time()
-    time.sleep(1)
+    if time.time() - new_time >= 0.5:
+        # Detect if we're in a time monotonically increasing
+        # layer; if so, we don't need to sleep
+        time.sleep(0.5)
```

Comment:
I don't understand. For me, `time.time() - new_time >= 0.5` is usually false when using real time.

Anyway, I think we always want to sleep. At least more than the resolution of the FS/kernels we want to support. Or is it for another reason (I admit I don't know blobs and blob tests enough) ?

Reply:
Sleeping was done just to advance the `time.time()` counter, because that's what we create TIDs and pack based on. I don't *think* filesystem times are relevant at all. If the `time.time()` counter is guaranteed to have advanced by the pack resolution, we don't need to sleep.

Reply:
> You didn't answer about the fact that the condition (>= 0.5) is inconsistent with the comment.

True, I just fixed it 😄